ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -378,7 +378,7 @@ private void configureNatEnvironment() {
             .get()
             .requestPortForward(
                 this.config.getDiscovery().getBindPort(),
-                UpnpNatManager.Protocol.TCP,
+                UpnpNatManager.Protocol.UDP,
```

Comment:
Do we need both TCP and UDP or is discovery UDP only?

Reply:
Discovery is UDP only.  There's another request for rlpx TCP port below.